### ISSUE TITLE
Indent option for break in switch

### ISSUE THREAD

**Issue by Zamralik** [eslint/eslint#13817]
**What rule do you want to change?**

indent

**Does this change cause the rule to produce more or fewer warnings?**

Both

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option "SwitchBreak"

**Please provide some example code that this change will affect:**

```js
switch (my_var)
{
    case 'foo':
        doSomething();
    break;

    case 'bar':
        doSomethingElse();
    break;

    default:
        throw new Error("Invalid value");
}
```

**What does the rule currently do for this code?**

The break being on the same indent level as the case is an error.

**What will the rule do after it's changed?**

The break being on the same indent level as the case is valid for "SwitchBreak": 1.

**Are you willing to submit a pull request to implement this change?**

No, I do not understand how ESLint works internally

**Comment by mdjermanovic** [comment]
Thanks for the issue!

The `indent` rule is already very complex so it isn't likely we will be adding any more options for specific preferences.

You can set `"ignoredNodes"` option, which can be used to relax the rule if you disagree with the indentation in some cases.

With the following configuration, the code example will be lint-free:

```js
/*eslint indent: ["error", 4, { SwitchCase: 1, ignoredNodes: ["SwitchCase > BreakStatement"] }]*/

switch (my_var)
{
    case 'foo':
        doSomething();
    break;

    case 'bar':
        doSomethingElse();
    break;

    default:
        throw new Error("Invalid value");
}
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgaW5kZW50OiBbXCJlcnJvclwiLCA0LCB7IFN3aXRjaENhc2U6IDEsIGlnbm9yZWROb2RlczogW1wiU3dpdGNoQ2FzZSA+IEJyZWFrU3RhdGVtZW50XCJdIH1dKi9cblxuc3dpdGNoIChteV92YXIpXG57XG4gICAgY2FzZSAnZm9vJzpcbiAgICAgICAgZG9Tb21ldGhpbmcoKTtcbiAgICBicmVhaztcblxuICAgIGNhc2UgJ2Jhcic6XG4gICAgICAgIGRvU29tZXRoaW5nRWxzZSgpO1xuICAgIGJyZWFrO1xuXG4gICAgZGVmYXVsdDpcbiAgICAgICAgdGhyb3cgbmV3IEVycm9yKFwiSW52YWxpZCB2YWx1ZVwiKTtcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

I'm closing this issue per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies) about stylistic rules.

**Comment by Zamralik** [comment]
@mdjermanovic Thanks for the ignoredNodes rule that let me move the break statement on the same level as the case.
